### DISCUSSION TITLE
ImageGlass 9.2.0.1208 - Maroon 4

### DISCUSSION THREAD

**Discussion by d2phap** [d2phap/ImageGlass#2053] [General]
You can support this project by making a donation or purchasing the Microsoft Store version:
<a href="https://apps.microsoft.com/detail/9N33VZK3C7TH?launch=true&cid=GitHubRelease&mode=full">
  <img height="58" src="https://github.com/d2phap/ImageGlass/assets/3154213/08a071bb-a6ae-420c-b53b-2317004570d4" />
</a> [![Downloads](https://img.shields.io/github/downloads/d2phap/imageglass/9.2.0.1208/total?color=%232A7C91&label=downloads&style=for-the-badge)](https://imageglass.org/release/imageglass-9-2-0-1208-56)

### 🌱 New Features
- Added a new tool: Resize Image (#1458)
  + Access via _Main Menu > Tools > Resize image_, or <kbd>Alt+R</kbd>
  + Compatible with single-frame, non-animated image formats
  + Supports scaling from 1px to 16384px
  + Supports variety of resampling methods powered by [PhotoSauce.MagicScaler](https://photosauce.net) library
  + **Note:** Similar to other editing features, saving the resized image does not include its metadata such as exif profile.
- Crop tool v2 (#1795, #1882, #2027, #2028):
  + Updates the mouse cursor when Crop tool is open
  + Selection now aligns with the image source
  + Zooming/panning doesn't affect selection
  + Supports resizing the selection by dragging any edge, not just anchor points
  + Supports panning image with middle mouse during cropping
  + Supports snap-to-pixel selection with pixel-level precision
  + Selection stays intact when clicking outside it
  + Default selection respects Crop settings and Aspect Ratio
- Slideshow mode: Added Custom zoom (<kbd>Z</kbd>) feature (#2021)
- Custom toolbar button: Added support for hotkeys
- Added new API: `IG_ApplySettings(string json)` to apply settings from the Settings window via JSON format (#2010, #2042)

### 🔮 Improvements
- Updated [Magick.NET to v14.2](https://github.com/dlemstra/Magick.NET/releases/tag/14.2.0)
- Made WebView2 Runtime optional:
  + Launch ImageGlass without bypassing the Quick Setup dialog
  + Quick Setup displays a warning with unavailable features and a download link (#2029)
  + SVG viewing reverts to the native renderer
  + About and Check for Update dialogs use the native Task Dialog
  + Settings opens the `igconfig.json` file in the associated app (or Notepad)
  + MSI installer: Remove WebView2 Runtime check
- Maximum image dimensions, for those images larger than 16,384 pixels:
  + Removed the limit on maximum image dimensions
  + Supports rendering images in full resollution (#1863, #1949)
  + Uses Direct2D software rendering
- Added support for automatically sorts new images based on the current sort order (#1467)
- Removed alert sound on image deletion confirmation popup (#1997)
- Allows user to set slideshow interval less than 1 second (#2047)
- Added support for app protocol for Custom Toolbar buttons, Tools, and Edit apps (#2002)
  + Enter the app protocol in the Executable box (must end with `:`)
  + Use the Argument box for parameters; `<file>` passes the current image path
  + Verify your setting in the "Command preview"
- Added a "Learn more" link to image interpolation settings, thanks to @stevencaiOR (#1982)
- Improved the method for retrieving image files in the directory
- Enabled dark mode for tooltips in the Crop Tool and Color Picker
- Enabled dark mode and accent color for the progress bar in the Export Frame dialog

### 🐛 Bugfixes
- Fixed hotkey parsing issues for non-English languages (#1988)
- Fixed malformed characters displayed for hotkeys in non-English languages (#1999)
- Fixed layout issues in the Settings window caused by a new Microsoft Edge version
- Fixed the Settings window staying inactive when reopened
- Fixed an issue where SVG image were blinking when opened with the WebView2 option enabled
- Fixed the duplication of values in the Arguments field of the "ToolbarButtons" setting on each app launch

## 🟣 Other Updates
- Published [ExifGlass app v1.7](https://github.com/d2phap/ExifGlass)
- Published [ImageGlass.Tools 1.9200.0](https://github.com/ImageGlass/ImageGlass.Tools)

## ⭐ New Contributors
* @stevencaiOR made their first contribution in https://github.com/d2phap/ImageGlass/pull/1982

**Full Changelog**: https://github.com/d2phap/ImageGlass/compare/9.1.8.1104...9.2.0.1208

<hr /><em>This discussion was created from the release <a href='https://github.com/d2phap/ImageGlass/releases/tag/9.2.0.1208'>ImageGlass 9.2.0.1208 - Maroon 4</a>.</em>

**Comment by leo-smi** [comment]
Thank you for resize feature!
